File: app/src/main/java/com/example/amprime/firebaseauth/Retrofit/RetrofitClient.java
package com.example.amprime.firebaseauth.Retrofit;


import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by amprime on 11/14/17.
 */

public class RetrofitClient {
    private static Retrofit retrofit = null;
    public static Retrofit getRetrofit(String BaseUrl){
        if(retrofit==null){
            retrofit = new Retrofit
                    .Builder()
                    .baseUrl(BaseUrl)
                    .addConverterFactory(GsonConverterFactory.create()).build();

        }
        return retrofit;
    }
}

File: app/src/main/java/com/example/amprime/firebaseauth/retrofit/RetrofitClient.kt
package com.example.amprime.firebaseauth.retrofit


import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

/**
 * Created by amprime on 11/14/17.
 */

object RetrofitClient {
    private var retrofit: Retrofit? = null
    fun getRetrofit(BaseUrl: String): Retrofit? {
        if (retrofit == null) {
            retrofit = Retrofit.Builder()
                    .baseUrl(BaseUrl)
                    .addConverterFactory(GsonConverterFactory.create()).build()

        }
        return retrofit
    }
}